src/seedu/addressbook/Main.java

```diff
@@ -106,12 +106,18 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
+        command.setData(addressBook, lastShownList);
+        CommandResult result = command.execute();
         try {
-            command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        }
+        catch (StorageOperationException e) {
+            ui.showToUser(e.getMessage());
+            ui.showToUser("Please ensure storage file is not set to \"Read-only\"");
```

Comment:
Good job on printing a message to the user here. Maybe you could add the file location as well? 
For example: "Please ensure storage file `/path/to/file` is not set to read-only."